pkg/agent/agent.go

```diff
@@ -569,7 +561,30 @@ func (i *Initializer) initNodeLocalConfig() error {
 		return nil
 	}
 
-	// Spec.PodCIDR can be empty due to misconfiguration
+	// Parse all PodCIDRs first, so that we could support IPv4/IPv6 dual-stack configurations.
+	if node.Spec.PodCIDRs != nil {
+		for _, podCIDR := range node.Spec.PodCIDRs {
+			_, localSubnet, err := net.ParseCIDR(podCIDR)
+			if err != nil {
+				klog.Errorf("Failed to parse subnet from CIDR string %s: %v", node.Spec.PodCIDR, err)
+				return err
+			}
+			if localSubnet.IP.To4() != nil {
+				if i.nodeConfig.PodIPv4CIDR != nil {
+					return fmt.Errorf("one IPv4 PodCIDR is allowed at most")
```

Comment:
Not a big deal, but maybe no need to return an error, but continue the loop?

Reply:
So you mean even if there thre multiple IPv4 subnet CIDR(or multiple IPv6 subnets), we could only use the first one and ignore others?

Reply:
Yes. What you think?

Reply:
Make sense, then I will add a warning to ignore the later CIDR if the field is already configured.